lib/src/di/injector.dart

```diff
@@ -55,22 +64,28 @@ void injectDependencies(Environment flavor) {
   _locator.registerSingleton<AppInfoRepository>(AppInfoLocalRepository());
   _locator.registerSingleton<PersistenceRepository>(SharedPreferencesPersistenceRepository());
   _locator.registerSingleton<SettingsRepository>(SettingsLocalRepository(_locator<PersistenceRepository>()));
-  _locator.registerSingleton<PresentationShowingRepository>(PresentationShowingLocalRepository(_locator<PersistenceRepository>()));
+  _locator.registerSingleton<PresentationShowingRepository>(
+      PresentationShowingLocalRepository(_locator<PersistenceRepository>()));
+  _locator.registerLazySingleton<AuthenticationRepository>(() => FirebaseAuthenticationRepository());
   _locator.registerFactory<CachePolicy<Article>>(() => ArticleCachePolicy());
   _injectApiDependencies();
   _injectNotifiers();
 }
 
 void _injectApiDependencies() {
+  _locator.registerSingleton<ExceptionMapper>(ApiExceptionMapper(), instanceName: _Constants.apiExceptionMapper);
```

Comment:
w sumie chyba nie muszą być to singletony? 🤔

Reply:
a no nie